domino_data/auth.py

```diff
@@ -5,20 +5,41 @@
 from os.path import exists
 
 import attr
+import backoff
+import httpx
 from pyarrow import flight
 
 from datasource_api_client.client import Client
 
 
+@backoff.on_exception(backoff.expo, httpx.HTTPStatusError, max_time=2)
+def get_jwt_token(url: str) -> str:
+    """Gets a domino token from local sidecar API.
+
+    Args:
+        url: base url of sidecar container serving token API
+
+    Returns:
+        JWT as string
+
+    Raises:
+        HTTPStatusError: if the API returns an error
```

Comment:
did we test that if it doesn't exist it returns a HTTPStatusError? Not empty or the likes

Reply:
I'm adding tests next, not sure what you mean by "doesn't exists" though?

Reply:
seems to me the assumption this code is making is that the api will throw a 404 if valid token not found. fair assumption but it might be good to check that the api is not doing something dumb like returning empty token